ComponentKit/StatefulViews/CKStatefulViewComponentController.mm

```diff
@@ -131,13 +131,13 @@ - (void)_presentStatefulView
 
 - (void)_relinquishStatefulViewIfPossible
 {
-  if ([self canRelinquishStatefulView]) {
+  if ([self canRelinquishStatefulView] && _statefulView) {
     [[CKStatefulViewReusePool sharedPool]
      enqueueStatefulView:_statefulView
      forControllerClass:[self class]
      context:_statefulViewContext
      mayRelinquishBlock:^BOOL{
-       if (!self->_mounted && [self canRelinquishStatefulView]) {
+       if (self->_statefulView && !self->_mounted && [self canRelinquishStatefulView]) {
```

Comment:
Here's the issue: _relinquishStatefulViewIfPossible can be called multiple times. I missed that the previous version of this code was using the `if (_statefulView)` check as a way to guarantee that only one block won and actually enqueued the view into the reuse pool.

So what would happen here is that the stateful view could be enqueued into the reuse pool as many times as this method was called, which could result in a stateful view being used for more than one controller at a time.